### ISSUE TITLE
Deploy to bare metal

### ISSUE THREAD

**Issue by bachp** [NixOS/nixops#510]
It would be nice if NixOps could be used to deploy to bare metal target.

One possibility would be to build up some iPXE server that allows to boot minimal NixOS images over network. The server would also register all machines that booted from it with their IP and MAC. A new NixOps backend could then query this service and pick a machine from the pool, or take the one with a specified MAC address.

**Comment by domenkozar** [comment]
There are multiple ways to do that. It's hard to really provide a generic automated step, as it depends on hardware an use cases.

We could however keep a reference of different solutions, for example I use https://github.com/snabblab/snabblab-nixos/blob/master/make-iso.nix

**Comment by bachp** [comment]
A place where solutions for different hardware and use cases are documented would be really appreciated. I din't know about the link you posted :smile: 

In the long run there might be some parts that can be shared and build up some kind of framework that can be reuses in most cases.

**Comment by domenkozar** [comment]
Indeed. I might have this use case very soon again, so I'll see if I can refactor some parts of it :)

**Comment by domenkozar** [comment]
Indeed. I might have this use case very soon again, so I'll see if I can refactor some parts of it :)

**Comment by domenkozar** [comment]
Relevant: https://github.com/cleverca22/nix-tests/tree/master/kexec
